hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/StructuredTraceGraphBuilder.java

```diff
@@ -0,0 +1,47 @@
+package org.hypertrace.traceenricher.trace.util;
+
+import org.hypertrace.core.datamodel.StructuredTrace;
+import org.hypertrace.core.datamodel.shared.StructuredTraceGraph;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class StructuredTraceGraphBuilder {
+  private static final Logger LOG = LoggerFactory.getLogger(StructuredTraceGraphBuilder.class);
+
+  private static ThreadLocal<StructuredTraceGraph> cachedGraph = new ThreadLocal<>();
```

Comment:
I understand that it is a quick optimisation. My question is eventually should we instead of caching the StructuredTraceGraph at thread level, pass around the instance of this graph between enrichers? Anyways the graph is going to be changed for each Trace. 
Any enrichment task can then update the StructuredTraceGraph if there is a need. That way we also get rid of the need to compare the graphs.

Reply:
Currently, it is another way around, the `trace` is changing during the enrichment process while processing by going over every span and so we have to re-construct StructureTraceGraph every time if the `trace` state has been modified. Yes, we have to rethink the current processing, and if it's the right data structure, or can we do it differently.